up-streamer/src/ustreamer.rs

```diff
@@ -159,9 +169,27 @@ impl ForwardingListeners {
                     .register_listener(&any_uuri(), Some(&uauthority_to_uuri(out_authority)), forwarding_listener.clone()));
 
                 if let Err(err) = reg_res {
-                    warn!("{FORWARDING_LISTENERS_TAG}:{FORWARDING_LISTENERS_FN_INSERT_TAG} unable to register listener, error: {err}");
+                    warn!("{FORWARDING_LISTENERS_TAG}:{FORWARDING_LISTENERS_FN_INSERT_TAG} unable to register request listener, error: {err}");
                 } else {
-                    debug!("{FORWARDING_LISTENERS_TAG}:{FORWARDING_LISTENERS_FN_INSERT_TAG} able to register listener");
+                    debug!("{FORWARDING_LISTENERS_TAG}:{FORWARDING_LISTENERS_FN_INSERT_TAG} able to register request listener");
+                }
+
+                for (topic, subscribers) in initial_subscriber_cache {
+                    let mut authority_name_hash_set: HashSet<String> = HashSet::new();
```

Comment:
See my comment over in usubscription-static-file/src/lib.rs.

Generally speaking, I think we can probably be more efficient in how we store and look up to find an entry in the `SubscriptionCache` we will need.

Reply:
Tried for something more efficient, take a look